armi/reactor/reactors.py

```diff
@@ -2129,9 +2129,103 @@ def growToFullCore(self, cs):
 
-    def setPitchUniform(self, pitchInCm):
+    def checkGridPlateMaterialConsistency(self):
+        """check that the grid plates in the core are the same material"""
+        gridPlateBlocks = self.getBlocks(Flags.GRID_PLATE)
+        refGridPlateMat = (
+            self.getFirstBlock(Flags.GRID_PLATE).getComponent(Flags.GRID_PLATE).material
+        )
+        for b in gridPlateBlocks:
+            gridPlateComponent = b.getComponent(Flags.GRID_PLATE)
+            gridPlateMaterial = gridPlateComponent.material
+            if not isinstance(gridPlateMaterial, type(refGridPlateMat)):
+                runLog.warning(
+                    "Grid plate material for assembly is not consistent! Grid plate expansion/contraction assumes\n"
+                    f"a consistent material and will use {refGridPlateMat}. The inconsistency is:\n"
+                    f"Reference grid plate material = {refGridPlateMat}\n"
+                    f"Inconsistent grid plate material = {gridPlateMaterial} -- is in {b.parent}"
+                )
+
+    def calculateNewGridPlatePitchFromTemp(self, inletTemp: float) -> float:
+        """calculate a grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        inletTemp : float
+            an isothermal temperature in C to apply to cs["Tin"]
         """
-        set the pitch in all blocks
+        self.checkGridPlateMaterialConsistency()
+        # retrieve the first grid plate's material (all grid plates are the same material, so just get the first)
+        gridPlateMaterial = (
+            self.getFirstBlock(Flags.GRID_PLATE).getComponent(Flags.GRID_PLATE).material
+        )
+        currentGridPlatePitch = self.getAssemblyPitch()
+        currentGridPlateTemp = self.refAssem.p.THcoolantInletT
+        runLog.info(
+            f"Using a {Flags.GRID_PLATE} material of {gridPlateMaterial} for grid plate expansion/contraction calculations "
+            f"with an initial temperature of {currentGridPlateTemp} C and new temperature of {inletTemp} C."
+        )
+        # compute expansion/contraction fraction
+        dll = gridPlateMaterial.linearExpansionFactor(
+            Tc=inletTemp,
+            T0=currentGridPlateTemp,
+        )
+        # compute new grid place pitch
+        return currentGridPlatePitch * (1.0 + dll)
+
+    def updateGridPlatePitch(self, inletTemp: float, updateInletTemp: bool = False):
+        """update the grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        inletTemp : float
+            an isothermal core bulk average inlet temperature in C
+        updateInletTemp: bool, optional
+            optional boolean to determine if the bulk average core inlet temperature should be updated to inletTemp
+
+        Notes
+        -----
+        1) all grid plate blocks are the same material
+        2) all grid plate blocks are isothermal and equal to inletTemp
```

Comment:
SUPER minor thought.

Can we put the word "assumption" in these notes?  I just want to make it clear to some stranger 5 years from now that these are our simplifying assumptions.  

Sorry to be nit-picky, I'm just want it to be transparent.

Reply:
Great idea! I like it. 
https://github.com/terrapower/armi/pull/993/commits/43dea385bc0d24d075b241ab1e6399e1d9956a1f